### PR TITLE
Upgrade existing Leaflet Plugins

### PR DESCRIPTION
**Pull request by bhaskarvk**
@jcheng5 This PR is ready for review/merge.

All plugins have been upgraded and more functionality added. I'll update the gh-pages branch in a diff. PR. For now there are lot's of examples in the examples dir.

### CODE REVIEW

File: R/plugin-minimap.R [hunk 1]
```diff
@@ -14,60 +16,109 @@ leafletMiniMapDependencies <- function() {
...
 #'
 #' leaf <- leaflet() %>%
 #'   addTiles() %>%
 #'   addMiniMap()
 #'
+#' @seealso \code{\link{providers}}
 #' @export
 addMiniMap <- function(
   map,
+  tiles = NULL,
```
Comment (jcheng5) [inline R/plugin-minimap.R]: Is it important that this argument be first? This functionality has been on master for a long time, I hate to break backward compatibility.

File: inst/htmlwidgets/leaflet.js [hunk 2]
```diff
@@ -986,7 +986,13 @@ function addMarkers(map, df, group, clusterOptions, clusterId, markerFunc) {
     var clusterGroup = this.layerManager.getLayer("cluster", clusterId),
         cluster = clusterOptions !== null;
     if (cluster && !clusterGroup) {
-      clusterGroup = _leaflet2.default.markerClusterGroup(clusterOptions);
```
Comment (jcheng5) [inline inst/htmlwidgets/leaflet.js]: Note to self: Left off here

File: inst/htmlwidgets/plugins/Leaflet-MiniMap/Minimap-binding.js [hunk 3]
```diff
@@ -1,11 +1,24 @@
-  LeafletWidget.methods.addMiniMap = function(position, width, height , collapsedWidth,
```
Comment (jcheng5) [inline inst/htmlwidgets/plugins/Leaflet-MiniMap/Minimap-binding.js]: Indentation seems to have gotten a bit funky in this file

Comment (bhaskarvk) [inline inst/htmlwidgets/plugins/Leaflet-MiniMap/Minimap-binding.js]: ack!

Comment (bhaskarvk) [inline inst/htmlwidgets/plugins/Leaflet-MiniMap/Minimap-binding.js]: We are not going from 1.0.2 to 1.0.3, right ? So just add the new stuff under existing 1.0.2 ?

Comment (jcheng5) [inline inst/htmlwidgets/plugins/Leaflet-MiniMap/Minimap-binding.js]: Let's change DESCRIPTION to 1.0.2.9000, and create a new 1.1 item in NEWS. When you believe you have a build that's ready for CRAN, then change to 1.1 in DESCRIPTION.

**Comment by bhaskarvk** [comment]
I've implemented all the changes and now merging with trunk.

**Pull request merged**